modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -793,6 +792,10 @@ public Set<String> dataNodes(int zoneId) {
         });
     }
 
+    public CompletableFuture<Set<String>> versionedDataNodes(int zoneId, long revision) {
```

Comment:
- I'd rather user `dataNodes` instead of `versionedDataNodes`
- It's not revision it's causalityToken.
- causalityToken should be the first parameter, see `org.apache.ignite.internal.table.distributed.TableManager#tableAsync(long, java.util.UUID)` or `org.apache.ignite.internal.index.IndexManager#createIndexLocally` for example.
- Where's the javadoc?

Reply:
Fixed. I think that the java doc will be changed with more details.